benchmarks/microbenchmarks/benchmark_inference.py

```diff
@@ -0,0 +1,74 @@
+"""
+Inference benchmark runner
+
+This script runs inference benchmarks and generates a micro-benchmarking report for it.
+- run() function is the main entry point for running inference benchmarks.
+"""
+
+from copy import deepcopy
+from pathlib import Path
+from typing import Dict
+
+import torch
+from utils import (
+    BenchmarkConfig,
+    benchmark_model_inference_in_microseconds,
+    clean_caches,
+    create_model_and_input,
+    quantization_string_to_quantization_config,
+)
+
+from torchao.quantization import quantize_
+
+
+def run(config: BenchmarkConfig) -> Dict[str, float]:
+    """Run inference benchmarks"""
+    clean_caches()  # Clean caches
+
+    # Create output directory if it doesn't exist
+    Path(config.output_dir).mkdir(parents=True, exist_ok=True)
+
+    base_model, input_data = create_model_and_input(
+        config.model_type,
+        config.m,
+        config.k,
+        config.n,
+        high_precision_dtype=config.high_precision_dtype,
+        device=config.device,
+    )
+
+    # Use quantize_ to apply each quantization function to the model
+    m_copy = deepcopy(base_model).eval().to(config.device)
+    quantization_config = quantization_string_to_quantization_config(
+        config.quantization, high_precision_dtype=config.high_precision_dtype
+    )
+    if quantization_config:
+        quantize_(m_copy, quantization_config)
+    if config.compile is not "false":
+        print("Compiling model....")
+        if compile == "true":
+            m_copy = torch.compile(m_copy, fullgraph=True)
+        else:
+            m_copy = torch.compile(m_copy, mode=config.compile, fullgraph=True)
+
+    # Run benchmarks
+    result = {**config.to_dict()}
+
+    # Benchmark time to run an inference call for quantized model
+    model_time = benchmark_model_inference_in_microseconds(
+        model=m_copy, input_data=input_data
+    )
+    result["benchmark_model_inference_in_microseconds"] = model_time
```

Comment:
nit: "benchmark_model_inference_in_microseconds" sounds like a function name, I think a name such as "model_inference_time_ms" is more descriptive of this value